Tests/UnitTests/Mocks/MockDeviceCache.swift

```diff
@@ -221,4 +221,25 @@ class MockDeviceCache: DeviceCache {
         invokedClearLatestNetworkAndAdvertisingIdsSentParameters = (appUserID, ())
         invokedClearLatestNetworkAndAdvertisingIdsSentParametersList.append((appUserID, ()))
     }
+
+    var invokedSetLatestNetworkAndAdvertisingIdsSent = false
+    var invokedSetLatestNetworkAndAdvertisingIdsSentCount = 0
+    var invokedSetLatestNetworkAndAdvertisingIdsSentParameters:
+        (adIdsByNetwork: [AttributionNetwork: String], appUserID: String?)?
+    var invokedSetLatestNetworkAndAdvertisingIdsSentParametersList =
+        [(adIdsByNetwork: [AttributionNetwork: String], appUserID: String?)?]()
```

Comment:
Does this need to be an array of optionals? I think we always append non-nil:
```suggestion
        [(adIdsByNetwork: [AttributionNetwork: String], appUserID: String?)]()
```